example/example20.cpp

```diff
@@ -0,0 +1,147 @@
+/*
+  example/example20.cpp -- Usage of structured numpy dtypes
+
+  Copyright (c) 2016 Ivan Smirnov
+
+  All rights reserved. Use of this source code is governed by a
+  BSD-style license that can be found in the LICENSE file.
+*/
+
+#include "example.h"
+
+#include <pybind11/numpy.h>
+#include <cstdint>
+#include <iostream>
+
+namespace py = pybind11;
+
+struct SimpleStruct {
+    bool x;
+    uint32_t y;
+    float z;
+};
+
+std::ostream& operator<<(std::ostream& os, const SimpleStruct& v) {
+    return os << "s:" << v.x << "," << v.y << "," << v.z;
+}
+
+struct PackedStruct {
+    bool x;
+    uint32_t y;
+    float z;
+} __attribute__((packed));
+
+std::ostream& operator<<(std::ostream& os, const PackedStruct& v) {
+    return os << "p:" << v.x << "," << v.y << "," << v.z;
+}
+
+struct NestedStruct {
+    SimpleStruct a;
+    PackedStruct b;
+} __attribute__((packed));
```

Comment:
Oh, good catch, I'll fix that tonight and open a PR to pybind11 :)

// please see my other few comments above